quinn/src/send_stream.rs

```diff
@@ -43,53 +43,80 @@ impl SendStream {
         }
     }
 
-    /// Write bytes to the stream
+    /// Attempts to write a buffer into this stream, returning how many bytes were written
     ///
-    /// Yields the number of bytes written on success. Congestion and flow control may cause this to
-    /// be shorter than `buf.len()`, indicating that only a prefix of `buf` was written.
+    /// Unless this method errors, it waits until some amount of `buf` can be written into this
+    /// stream, and then writes as much as it can without waiting again. Due to congestion and flow
+    /// control, this may be shorter than `buf.len()`. On success this resolves to the length of
+    /// the prefix that was written.
     ///
-    /// This operation is cancel-safe.
+    /// This method is cancellation safe: If it does not resolve, no bytes were written.
```

Comment:
nit: tokio uses a `# Cancel safety` heading for these.  Which makes it stand out still a bit more.  I'd be +1 to adopt that convention.